// @ahk-freiburg/hwrm-mapping/esm2020/lib/mapping.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MapWindowComponent } from './components/map-window/map-window.component';
import { MappingModuleConfigurationToken } from './mapping.module.configuration';
import * as i0 from "@angular/core";
export class MappingModule {
    static forRoot(config) {
        return {
            ngModule: MappingModule,
            providers: [
                {
                    provide: MappingModuleConfigurationToken,
                    useValue: config
                }
            ]
        };
    }
}
MappingModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MappingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MappingModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MappingModule, declarations: [MapWindowComponent], imports: [CommonModule], exports: [MapWindowComponent] });
MappingModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MappingModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MappingModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        MapWindowComponent
                    ],
                    exports: [
                        MapWindowComponent
                    ],
                }]
        }] });
