// @ahk-freiburg/hwrm-mapping/esm2020/lib/components/map-window/map-window.component.mjs
import { Component, ChangeDetectionStrategy, Inject } from '@angular/core';
import esriConfig from '@arcgis/core/config';
import WebMap from '@arcgis/core/WebMap';
import MapView from '@arcgis/core/views/MapView';
import { MappingModuleConfigurationToken } from '../../mapping.module.configuration';
import * as i0 from "@angular/core";
export class MapWindowComponent {
    constructor(config) {
        this.config = config;
    }
    ngOnInit() {
        // Consuming app can pass a local asset path otherwise assets will be
        // puled from CDN
        if (this.config.assetPath)
            esriConfig.assetsPath = this.config.assetPath;
        const webMap = new WebMap({
            basemap: "streets-vector"
        });
        const view = new MapView({
            container: 'map',
            map: webMap
        });
    }
}
MapWindowComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MapWindowComponent, deps: [{ token: MappingModuleConfigurationToken }], target: i0.ɵɵFactoryTarget.Component });
MapWindowComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.2.7", type: MapWindowComponent, selector: "hwrm-mapping-map-window", ngImport: i0, template: "<div id=\"map\"></div>", styles: ["#map{width:inherit;height:inherit}\n"], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MapWindowComponent, decorators: [{
            type: Component,
            args: [{ selector: 'hwrm-mapping-map-window', changeDetection: ChangeDetectionStrategy.OnPush, template: "<div id=\"map\"></div>", styles: ["#map{width:inherit;height:inherit}\n"] }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [MappingModuleConfigurationToken]
                }] }]; } });
